vividus-plugin-lighthouse/src/main/java/org/vividus/lighthouse/LighthouseSteps.java

```diff
@@ -232,6 +277,65 @@ private Map<String, BigDecimal> getMetrics(LighthouseResultV5 result)
         return metrics;
     }
 
+    private LighthouseResultV5 executePagespeedTest(String url, String strategy)
+            throws IOException
+    {
+        if (performanceScoreSelector == NONE)
+        {
+            return executePagespeed(url, strategy, true);
+        }
+
+        isTrue(categories.contains("performance"),
+                "The %s performance score selector requires 'performance' category to be present in scan parameters",
+                performanceScoreSelector);
+
+        List<LighthouseResultV5> results = new ArrayList<>(ATTEMPT_NUMBER);
+
+        LighthouseCategoryV5 previous = null;
+        for (int index = 0; index < ATTEMPT_NUMBER; index++)
+        {
+            while (true)
```

Comment:
need to add extra condition to stop this infinite loop

Reply:
suggestions?